### PR TITLE
New Skim to widget

### PR DESCRIPTION
**Pull request by robert00s**
https://github.com/koreader/koreader/issues/2819#issuecomment-332261968
+ refactoring code in skimtowidget
+ add chapter markers
+ add next/prev chapter buttons
+ add next/prev bookmark buttons

screenshot 2017-10-21 17-41-26

### CODE REVIEW

**Comment by Frenzie** [comment]
Could you use nicer symbols like in the mockup?

* ☆ http://www.fileformat.info/info/unicode/char/2606/index.htm
* ◁ http://www.fileformat.info/info/unicode/char/25c1/index.htm
* ▷ http://www.fileformat.info/info/unicode/char/25b7/index.htm

(triangles also available in small)

**Comment by robert00s** [comment]
Of course :)

screenshot 2017-10-21 18-40-38

**Comment by Frenzie** [comment]
PS I didn't necessarily mean those specific symbols.

Btw, there are other symbols that might go with it better. I think the first is the one you can just type easily:

* | http://www.fileformat.info/info/unicode/char/7c/index.htm
* ❘http://www.fileformat.info/info/unicode/char/2758/index.htm
* ❚http://www.fileformat.info/info/unicode/char/275a/index.htm

Or there might be integrated first/last arrows too? It was just a little suggestion to get started really. :-)

File: frontend/apps/reader/skimtowidget.lua [hunk 1]
```diff
@@ -58,63 +59,72 @@ function SkimToWidget:init()
...
-    }
-    local progress_bar = ProgressWidget:new{
+    self.progress_bar = ProgressWidget:new{
         width = self.screen_width * 0.9,
         height = Screen:scaleBySize(30),
         percentage = self.curr_page / self.page_count,
-        ticks = nil,
-        last = nil,
+        ticks = self.ticks_candidates,
+        tick_width = 1,
```
Comment (Frenzie) [inline frontend/apps/reader/skimtowidget.lua]: Looks like the check is correct about this needing to be scaleBySized? Might consider doing Size.line.thin.

**Comment by AlanSP1** [comment]
Very nice addition. 👍

**Comment by robert00s** [comment]
I used | http://www.fileformat.info/info/unicode/char/7c/index.htm earlier.

**Comment by Frenzie** [comment]
@robert00s Did you forget to commit something or what do you mean exactly?

**Comment by robert00s** [comment]
I mean that in that lines
https://github.com/koreader/koreader/pull/3389/files#diff-aa5ff64a3eb7439431a2389beafd87a9R198
https://github.com/koreader/koreader/pull/3389/files#diff-aa5ff64a3eb7439431a2389beafd87a9R216
I used this symbol http://www.fileformat.info/info/unicode/char/7c/index.htm from the first commit so I didn't have to change it.

**Comment by Frenzie** [comment]
Did you try any alternatives? I don't like the alignment. :-/ I'm really just giving a few suggestions to get started.

I don't just mean different lines. Also potentially different arrows, such as ⇾, although the triangle is the one I opted for at this point in time.

The left and right vertical box lines could be nice if they're in the font:
⎸ http://www.fileformat.info/info/unicode/char/23b8/index.htm
⎹ http://www.fileformat.info/info/unicode/char/23b9/index.htm

**Comment by Frenzie** [comment]
PS Of course we could always construct something à la https://github.com/koreader/koreader/blob/23ab33c0b98b4079a903a741f6ea9eb66c199eb3/frontend/ui/widget/checkmark.lua#L30-L50

**Comment by robert00s** [comment]
Better? :)
screenshot 2017-10-22 12-14-52

**Comment by Frenzie** [comment]
Alignment yes, but could you change it back to the other triangle and star? Otherwise the amount of black is visually a bit distracting. (Also see the dictionary.)

**Comment by robert00s** [comment]
screenshot 2017-10-22 12-23-55

**Comment by Frenzie** [comment]
Perfect! (Or at least close enough. =P)

**Pull request merged**